sql/table.go

```diff
@@ -300,29 +300,45 @@ func encodeSecondaryIndexes(tableID structured.ID, indexes []structured.IndexDes
 	return secondaryIndexEntries, nil
 }
 
-// TODO(tamird): make this not panic. Not critical, since a panic here
-// will just tank a single goroutine on the server and be silently
-// swallowed.
 func prepareVal(col structured.ColumnDescriptor, val parser.Expr) (interface{}, error) {
 	if val == parser.DNull {
 		return nil, nil
 	}
 
+	valTypeOk := true
 	switch col.Type.Kind {
 	case structured.ColumnType_BIT:
-		return bool(val.(parser.DBool)), nil
+		if v, ok := val.(parser.DBool); ok {
+			return bool(v), nil
+		}
+		valTypeOk = false
 	case structured.ColumnType_BOOL:
-		return bool(val.(parser.DBool)), nil
+		if v, ok := val.(parser.DBool); ok {
+			return bool(v), nil
+		}
+		valTypeOk = false
 	case structured.ColumnType_INT:
-		return int64(val.(parser.DInt)), nil
+		if v, ok := val.(parser.DInt); ok {
+			return int64(v), nil
+		}
+		valTypeOk = false
 	case structured.ColumnType_FLOAT:
-		return float64(val.(parser.DFloat)), nil
+		if v, ok := val.(parser.DFloat); ok {
+			return float64(v), nil
+		}
+		valTypeOk = false
 	// case structured.ColumnType_DECIMAL:
 	// case structured.ColumnType_DATE:
 	// case structured.ColumnType_TIME:
 	// case structured.ColumnType_TIMESTAMP:
 	case structured.ColumnType_CHAR, structured.ColumnType_TEXT, structured.ColumnType_BLOB:
-		return string(val.(parser.DString)), nil
+		if v, ok := val.(parser.DString); ok {
+			return string(v), nil
+		}
+		valTypeOk = false
+	}
+	if !valTypeOk {
+		return nil, fmt.Errorf("column type %s doesn't match value type %T", col.Type.Kind, val)
 	}
-	return nil, fmt.Errorf("Unsupported type: %T", val)
+	return nil, fmt.Errorf("unsupported type: %T", val)
```

Comment:
You can get rid of `valTypeOk` if you add a `default` case statement to the switch.